lib/ash_authentication_phoenix/strategy_router.ex

```diff
@@ -0,0 +1,50 @@
+defmodule AshAuthenticationPhoenix.StrategyRouter do
+  @behaviour Plug
+
+  def init(opts), do: opts
+
+  def call(conn, opts) do
+    opts
+    |> routes()
+    |> Enum.reduce_while(
+      {:not_found, conn},
+      fn {resource, strategy, path, phase}, {:not_found, conn} ->
```

Comment:
So from what I can tell, everything in this PR should work, but this part needs to be figured out still. You mentioned reusing the auth plug or auth plug logic from `AshAuthentication`, but that may be heavy handed as it requires the user to define it, so they'd need both an `AuthController` and an `AuthPlug` and that feels less than ideal.

However, the one change this new setup needs is that they have to provide a `not_found_plug` because we aren't just falling through their router anymore.

If we're cool with that, then I think we can just match on the route here and be done with it? Unless reusing the auth plug has some other tangible value.